pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -208,6 +222,40 @@ func watchCerts(ctx context.Context, certsDirs []string, watchFileEventsFn watch
 	watchFileEventsFn(ctx, fw.Event, minDelay, updateFunc)
 }
 
+// waitForCertsPresent is a blocking function that periodically checks the existence
+// of the certs, once every interval. It returns nil when all cert files are found
+// and non-empty, and returns error if the timeout is reached.
+func waitForCertsPresent(certs []CertSource, interval time.Duration, timeout time.Duration) (err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+	for _, cert := range certs {
+		for _, file := range cert.Files {
+			filename := path.Join(cert.Directory, file)
+			g.Go(func() error {
+				for {
+					fi, err := os.Stat(filename)
+					if err != nil {
```

Comment:
nit: this should probably use https://golang.org/pkg/os/#IsNotExist to avoid retrying on more permanent errors (e.g. permissions)

Reply:
Sure.